talk/expert/modules.tex

```diff
@@ -0,0 +1,643 @@
+  \vspace{-5mm}
+  \begin{columns}[t]
+    \begin{column}{.45\textwidth}
+      \begin{exampleblock}{Implementation unit foo}
+        \begin{cppcode*}{gobble=6}
+          module math:foo;
+
+          S f() { ... }
+        \end{cppcode*}
+      \end{exampleblock}
+    \end{column}
+    \begin{column}{.45\textwidth}
+      \begin{exampleblock}{Implementation unit bar}
+        \begin{cppcode*}{gobble=6}
+          module math:bar;
+
+          std::string b() { ... }
+        \end{cppcode*}
+      \end{exampleblock}
+    \end{column}
+  \end{columns}
+\end{frame}
+
+\begin{frame}[fragile,shrink=5]
+  \frametitlecpp[20]{Submodules and re-exporting}
+  \begin{block}{}
+    Modules can (re-)export other modules:
+  \end{block}
+  \vspace{-5mm}
+  \begin{columns}[t]
+    \begin{column}{.45\textwidth}
+      \begin{exampleblock}{Module base}
+        \begin{cppcode*}{gobble=6}
+          export module base;
+
+          export struct S { ... };
+        \end{cppcode*}
+      \end{exampleblock}
+    \end{column}
+    \begin{column}{.45\textwidth}
+      \begin{exampleblock}{Module bar}
+        \begin{cppcode*}{gobble=6}
+          export module hi;
+          import <string>;
+
+          export std::string b()
+            { ... }
+        \end{cppcode*}
+      \end{exampleblock}
+    \end{column}
+  \end{columns}
+  \begin{columns}[t]
+    \begin{column}{.45\textwidth}
+      \begin{exampleblock}{Module foo}
+        \begin{cppcode*}{gobble=6}
+          export module foo;
+          import export module base;
+          import bar;
+
+          export S f() { ... }
+        \end{cppcode*}
+      \end{exampleblock}
+    \end{column}
+    \begin{column}{.45\textwidth}
+      \begin{exampleblock}{Module math}
+        \begin{cppcode*}{gobble=6}
+          export module math;
+          export import foo;
+        \end{cppcode*}
+      \end{exampleblock}
+    \end{column}
+  \end{columns}
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitlecpp[23]{Standard library modules}
+  % see: Minimal module support for the standard library http://wg21.link/P2412
+  \begin{block}{Standard library modules}
+    \begin{itemize}
+      \item Use \mintinline{cpp}|import std;| for the entire C++ standard library. Basically everything in namespace \mintinline{cpp}|std|.
+      \begin{itemize}
+        \item E.g.\ \mintinline{cpp}{memcpy(a, b, n);}, \mintinline{cpp}{sqrt(val);}, \mintinline{cpp}{uint32_t i;} will not compile. You need to prefix \mintinline{cpp}{std::}.
+      \end{itemize}
+      \item Use \mintinline{cpp}|import std.compat;| for the entire C and C++ standard library
+      \item No macros are exported. For these, you still need to include the corresponding headers.
+      \begin{cppcode*}{gobble=4}
+        import std;        // everything C++
+        #include <cassert> // for assert()
+        #include <climits> // for e.g. CHAR_BIT
+      \end{cppcode*}
+      \item Further modularization of the standard library is in progress
```

Comment:
That's a contradiction to "for the entire C++ standard library" if it's still in progress.

Reply:
Then I need to rephrase. There is progress on providing more modules of the standard library. So far only `std` and `std.compat` have been standardized. We may see more modules like: `containers`, `fundamentals`, 'network' etc. in the future.

Reply:
It now reads: "Definition of further, smaller modules of the standard library is in progress (e.g. fundamentals, containers, networking, etc.)"